server/service/logging_campaigns.go

```diff
@@ -0,0 +1,79 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"github.com/kolide/fleet/server/contexts/viewer"
+	"github.com/kolide/fleet/server/kolide"
+	"github.com/kolide/fleet/server/websocket"
+)
+
+func (mw loggingMiddleware) NewDistributedQueryCampaign(ctx context.Context, queryString string, hosts []uint, labels []uint) (*kolide.DistributedQueryCampaign, error) {
+	var (
+		loggedInUser = "unauthenticated"
+		metrics      *kolide.TargetMetrics
+		err          error
+	)
+	if vc, ok := viewer.FromContext(ctx); ok {
+
+		loggedInUser = vc.Username()
+	}
+	metrics, err = mw.Service.CountHostsInTargets(ctx, hosts, labels)
```

Comment:
I'm cautious of doing too much work in the logging middleware. What if we added `kolide.TargetMetrics` as a field on the `DistributedQueryCampaign` struct and then populated it within the service methods that create campaigns? Then the metrics values could be used here and potentially by clients that receive the results.

Reply:
I updated to use the service instead of the logging middleware, I think this was a good change